SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularitySlaveAndRackManager.java

```diff
@@ -199,13 +204,22 @@ public void slaveLost(SlaveID slaveIdObj) {
 
     if (slave.isPresent()) {
       slaveManager.changeState(slave.get(), MachineState.DEAD, Optional.<String> absent(), Optional.<String> absent());
+      if (configuration.getDisasterDetection().isEnabled()) {
+        updateDisasterCounter(slave.get());
+      }
 
       checkRackAfterSlaveLoss(slave.get());
     } else {
       LOG.warn("Lost a slave {}, but didn't know about it", slaveId);
     }
   }
 
+  private void updateDisasterCounter(SingularitySlave slave) {
+    if (slave.getCurrentState().getState() == MachineState.ACTIVE) {
```

Comment:
I might not understand this part fully, but if the slave state is changed to `MachineState.DEAD`, will this ever be true?